php/class-media.php

```diff
@@ -1156,23 +1159,29 @@ public function get_cloudinary_id( $attachment_id ) {
 	 *
 	 * @param int $attachment_id The ID to get Cloudinary id for.
 	 *
-	 * @return string|null the ID or null if not existing.
+	 * @return string|false the ID or null if not existing.
 	 */
 	public function cloudinary_id( $attachment_id ) {
+		static $cloudinary_ids = array();
+
+		// Return cached ID if we've already gotten it before.
+		if ( isset( $cloudinary_ids[ $attachment_id ] ) ) {
+			return $cloudinary_ids[ $attachment_id ];
+		}
 
 		if ( ! $this->is_media( $attachment_id ) ) {
+			$cloudinary_ids[ $attachment_id ] = null;
 			return null;
```

Comment:
@DavidCramer 

We are returning `null`. I wonder if this has an impact on the issues we're seeing.